### PR TITLE
SPR-15604 Fixed cron expression validation method in CronSequenceGenerator

### PR DESCRIPTION
**Pull request by Enigo**
This method validated only fields length of the given expression, not the validity of cron expression. It seems logical that the method with name 'isValidExpression' would validate cron as well.
Plus, possible NPE fixed.

### CODE REVIEW

**Comment by snicoll** [comment]
@Enigo please create a Jira issue for this change.

**Comment by Enigo** [comment]
@snicoll, hey, [issue](https://jira.spring.io/browse/GREENHOUSE-658) raised, but I accidentally miss-clicked the project :( Is it possible to move the issue to the Spring Framework (SPR)?

**Comment by snicoll** [comment]
nope, I have no right on that project, sorry.

**Comment by Enigo** [comment]
Great, thanks! [Issue](https://jira.spring.io/browse/SPR-15604) raised)

File: spring-context/src/main/java/org/springframework/scheduling/support/CronSequenceGenerator.java [hunk 1]
```diff
@@ -260,6 +260,9 @@ private void reset(Calendar calendar, List<Integer> fields) {
 	 * Parse the given pattern expression.
 	 */
 	private void parse(String expression) throws IllegalArgumentException {
+		if (expression == null) {
```
Comment (deodeveloper) [inline spring-context/src/main/java/org/springframework/scheduling/support/CronSequenceGenerator.java]: Using standard way of spring exceptions being thrown. 
How about ```Assert.isNull(expression, "Cron expression cannot be null!")``` ?

Comment (Enigo) [inline spring-context/src/main/java/org/springframework/scheduling/support/CronSequenceGenerator.java]: I'm sure you meant `Assert.notNull` :) fixed.

**Comment by jhoeller** [comment]
Implemented in a slightly different fashion, as mentioned on the JIRA issue.

**Pull request closed without merging**